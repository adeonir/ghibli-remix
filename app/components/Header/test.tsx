import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import { Header, Props } from '.'

const props: Props = {
  film: {
    id: 'id',
    characters: [],
    comments: [],
    description: 'description',
    image: 'image',
    movie_banner: 'movie_banner',
    original_title: 'original_title',
    people: ['PersonName'],
    release_date: 'release_date',
    rt_score: '75',
    running_time: 'running_time',
    title: 'title',
  },
}

describe('<Header />', () => {
  it(`should render correctly`, () => {
    const { container } = render(<Header {...props} />, {
      wrapper: MemoryRouter,
    })

    expect(screen.getByRole(`heading`, { name: /title/i })).toBeInTheDocument()
    expect(screen.getByText(/title/i)).toBeInTheDocument()
    expect(screen.getByText(/description/i)).toBeInTheDocument()
    expect(screen.getByText(/release_date/i)).toBeInTheDocument()
    expect(screen.getByText(/running_time/i)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
