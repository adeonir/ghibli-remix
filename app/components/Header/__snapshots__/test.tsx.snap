// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<div>
  <div
    class="grid grid-cols-1 gap-8 lg:grid-cols-2"
  >
    <div
      class="space-y-4"
    >
      <h2
        class="text-4xl font-bold text-teal-500"
      >
        title
      </h2>
      <div
        class="grid-col-1 grid gap-4 sm:grid-cols-3"
      >
        <div>
          <span
            class="mb-1 flex border-b border-b-slate-500 text-sm text-slate-500"
          >
            Release date
          </span>
          <p
            class="text-lg text-slate-200"
          >
            release_date
          </p>
        </div>
        <div>
          <span
            class="mb-1 flex border-b border-b-slate-500 text-sm text-slate-500"
          >
            Time
          </span>
          <p
            class="text-lg text-slate-200"
          >
            running_time
            min
          </p>
        </div>
        <div>
          <span
            class="mb-1 flex border-b border-b-slate-500 text-sm text-slate-500"
          >
            Rating
          </span>
          <p
            class="text-lg text-slate-200"
          >
            7.5
            /10
          </p>
        </div>
      </div>
    </div>
    <div>
      <p
        class="text-base text-slate-100"
      >
        description
      </p>
    </div>
  </div>
</div>
`;
